app/main/services/query_builder.py

```diff
@@ -56,19 +56,19 @@ def construct_query(mapping, query_args, aggregations=[], page_size=100):
 
 
 def highlight_clause(mapping):
+    # Only fields with an entry in 'fields' will be returned as part of the highlighted key regardless of their value
+    # Here we attach all fields to take advantage of highlighting and html escaping.
     highlights = {
+        "number_of_fragments": 0,
+        "no_match_size": 500,
         "encoder": "html",
         "pre_tags": ["<mark class='search-result-highlighted-text'>"],
-        "post_tags": ["</mark>"]
-    }
-    highlights["fields"] = {}
-
-    # Get all fields searched and allow non-matches to a max of the searchSummary limit
-    for field in mapping.fields_by_prefix.get(mapping.text_search_field_prefix, ()):
-        highlights["fields"]["_".join((mapping.text_search_field_prefix, field))] = {
-            "number_of_fragments": 0,
-            "no_match_size": 500
+        "post_tags": ["</mark>"],
+        "fields": {
+            f"{mapping.text_search_field_prefix}_{field}": {}
+            for field in mapping.fields_by_prefix.get(mapping.text_search_field_prefix, ())
```

Comment:
We don't need a for loop here, we can use the globbing `f"{mapping.text_search_field_prefix}_*" : {}`

Reply:
Nice!!

Reply:
Updated 👍